### PR TITLE
[PropertyInfo] Add PhpStan extractor to PropertyInfo available extractors

### LINKED ISSUE #16730: [PropertyInfo] Add support for promoted properties in PhpStanExtractor
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46056
| PR author(s) | @simPod
| Merged in    | 6.1

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Resolves #16730

### CODE REVIEW

File: components/property_info.rst [hunk 1]
```diff
@@ -411,6 +411,37 @@ library is present::
...
+PhpStanExtractor
+~~~~~~~~~~~~~~~
+
+.. note::
+
+    This extractor depends on the `phpstan/phpdoc-parser`_ and
+    `phpdocumentor/reflection-docblock`_ libraries.
+
+This extractor fetches informations thanks to the PHPStan parser. This
+extractor gather information from annotations of the comment section
```
Comment (HeahDude) [inline components/property_info.rst]:
```suggestion
it gathers information from annotations of properties ...
```

File: components/property_info.rst [hunk 2]
```diff
@@ -411,6 +411,37 @@ library is present::
...
+of properties and methods, such as ``@var``, ``@param`` or ``@return``::
+
+    // Foo.php
```
Comment (HeahDude) [inline components/property_info.rst]:
```suggestion
    // src/Domain/Foo.php
```
?

File: components/property_info.rst [hunk 3]
```diff
@@ -411,6 +411,37 @@ library is present::
...
+    class Foo
+    {
+        private string $bar;
+
+        /**
+         * @param string $bar
+         */
+        public function __construct(string $bar = 'baz') {
```
Comment (HeahDude) [inline components/property_info.rst]: I don't see any value in annotating a typed parameter and property.
Maybe should you remove native types in order to let `@param` do the job?

Comment (alexandre-daubois) [reply components/property_info.rst]: Definitely!

**Comment by javiereguiluz** [comment]
Thanks Alexandre for this contribution and thanks Jules for the detailed review.

Please note that we merged this into 6.1 branch instead of 5.4 and we added the `versionadded` directive too while merging. Cheers!

**Pull request merged**